### PR TITLE
Ruby on Rails Capstone - E-Wallet App 🛒

### PR DESCRIPTION
**Pull request by Hamzaoutdoors**
### In this project capstone we accomplished the following :

- [x] Used `Postgres` as your database.
- [x] Used `devise` for authentication.
- [x] Validated all user input to make sure that anyone with bad intentions cannot compromise your app.
- [x] The project has been deployed and accessible online using `heroku`
- [x] Your database schema should reflect the following entities : 
  - `Categories`, `Activities` and `Users` entities
- [x] Write and run integration tests for all views
 <div align="center">
    <kbd>
    gif
    <p>23 Request Specs passed</p>
       </kbd>
</div>

### Project presentation video ⏺️

[Video presentation](https://www.loom.com/share/1663fe730930447fa8c36a08c5c38f92)

_Thank you in advance for your feedback 😄_

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "new-password", placeholder: "password", class: "mb-4 rounded-3 py-2 px-5" %>
+  </div>
+
+  <div class="field">
+    <%= f.password_field :password_confirmation, autocomplete: "new-password", placeholder: "confirm your password", class: "mb-4 rounded-3 py-2 px-5" %>
+  </div>
+
+  <div class="d-flex justify-content-center">
+    <%= f.submit "Sign up", class: "button is-success is-focused mr-2" %>
```
Comment (Dipeshtwis) [inline app/views/devise/registrations/new.html.erb]: - [ ] I can see you have done a great job in this project, but still, the design is not looking similar for the registration
 page.

GIven design | Your project design
:---:|:---:
Screenshot from 2022-01-14 02-19-29 | Screenshot from 2022-01-14 02-19-58

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <div class="field">
+        <%= f.check_box :remember_me %>
+        <%= f.label :remember_me %>
+      </div>
+    <% end %>
+
+    <div class="d-flex justify-content-center">
+      <%= f.submit "Log in", class: "button is-success is-focused mr-2" %>
+      <%= render "devise/shared/links" %>
+    </div>
```
Comment (Dipeshtwis) [inline app/views/devise/sessions/new.html.erb]: - [ ] Also, the design is not looking similar for the login page. I would recommend using the design, color, font similar to the project requirements.


GIven design | Your project design
:---:|:---:
Screenshot from 2022-01-14 02-56-40 | Screenshot from 2022-01-14 02-18-55

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Hamzaoutdoors, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: All tests are passing. :heavy_check_mark: 
:+1: The app is running fine and deployed online.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (Hamzaoutdoors) [reply app/views/devise/registrations/new.html.erb]: ✅ Thank you for your feedback

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (Hamzaoutdoors) [reply app/views/devise/sessions/new.html.erb]: ✅ Thank you for your feedback

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**